explorer/interpreter/type_checker.cpp

```diff
@@ -1073,12 +1085,25 @@ class TypeChecker::ConstraintTypeBuilder {
       : ConstraintTypeBuilder(arena, MakeSelfBinding(arena, source_loc)) {}
   ConstraintTypeBuilder(Nonnull<Arena*> arena,
                         Nonnull<GenericBinding*> self_binding)
-      : self_binding_(PrepareSelfBinding(arena, self_binding)),
+      : arena_(arena),
```

Comment:
Not in this change, but this seems to be getting worth moving longer methods to forward declarations instead of inline definitions... I was surprised to have to scroll down 300 lines to find arena_ being added.

Reply:
Acknowledged.